File: Sorting/SelectionSort/selectionSort.java
package Sorting.SelectionSort;

import java.util.Arrays;

public class selectionSort {
    public static void main(String[] args) {
        int[] arr = {90,54,634,234,45,123,-345,-234,0,-100};
        System.out.println(Arrays.toString(sort(arr)));
    }
    public static int[] sort(int[] arr){

        int len = arr.length - 1;

        int index = 0;
        int i = 0;

        while(i < len){
        
            int max = arr[0];
            for(int j = 0; j <= len; j++){
                if(max < arr[j]){
                    max = arr[j];
                    index = j;
                }
            }
            int temp = arr[len];
            arr[len] = max;
            arr[index] = temp;
            len--;
            index = 0;
        }
        
        return arr;
    }

}

File: SelectionSort.cpp
#include <bits/stdc++.h>
using namespace std;

int * selection(int[],int);

int main(){
    int n;
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }
    int * ans;
    ans = selection(arr,n);
    for(int i = 0; i < n; i++){
        cout << ans[i] << " ";
    }
    cout << endl;
}

int * selection(int arr[], int n){

    for(int i = 0; i < n; i++){
        int index = 0;
        int max = arr[0];
        for(int j = 0; j < n-i; j++){
            if(arr[j] > max){
                max = arr[j];
                index = j;
            }
        }
        int temp = arr[(n-1)-i];
        arr[(n-1)-i] = max;
        arr[index] = temp;
    }
    return arr;
}